### PR TITLE
Feat/geekout certs v2

### PR DESCRIPTION
**Pull request by yapyuyou**
In this PR:
- Update index to include geekout V2 certs
- Add reusable backgrounds

### CODE REVIEW

File: src/templates/index.tsx [hunk 1]
```diff
@@ -3,10 +3,12 @@ import { templates } from "./certificateOfAchievement";
...
 import { geekOut2021Template } from "./geekOut2021";
 import { covidLetterOfAppreciationTemplate } from "./covidAppreciationLetter";
+import { geekOutV2Template } from "./geekOutV2";
 
 export const registry: TemplateRegistry<any> = {
   CERTIFICATE_OF_ACHIEVEMENT: templates,
   GEEK_OUT_2020: geekOut2020Template,
   GEEK_OUT_2021: geekOut2021Template,
+  GEEK_OUT_V2: geekOutV2Template,
   COVID_LETTER_OF_APPRECIATION_2021: covidLetterOfAppreciationTemplate
```
Comment (simboonlong) [inline src/templates/index.tsx:13]: question: what's v2? is it `GEEK_OUT_2022`?

File: src/templates/geekOutV2/types.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { v2 } from "@govtechsg/decentralized-renderer-react-components";
+import { geekout } from "@govtechsg/oa-schemata";
+
+export type DocumentType = "champion" | "runner-up" | "completion";
```
Comment (simboonlong) [inline src/templates/geekOutV2/types.tsx:4]: suggestion (non-blocking): do u want to add the type at oa schemata end?

File: src/templates/geekOutV2/geekOutV2.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    right: 25px;
+    width: 380px;
+    text-align: center;
+    color: #ffffff;
+    font-size: 18px;
+  }
+`;
+
+const getBackgroundImage = (documentType: DocumentType): any => {
+  switch (documentType) {
```
Comment (simboonlong) [inline src/templates/geekOutV2/geekOutV2.tsx:100]: looks like duplicated from https://github.com/Open-Attestation/govtech-decentralized-renderer/blob/master/src/templates/geekOut2021/geekOut2021.tsx#L60, refactor to utils?

File: src/templates/index.tsx (hunk 1 above)
Comment (yapyuyou) [reply src/templates/index.tsx:13]: Yeah for this year, but the intention is that we don't have to create a new type every year

Comment (simboonlong) [reply src/templates/index.tsx:13]: oh so 2021 and 2020 are "V1". quick check wats the diff in v2? mainly aesthetics?

File: src/templates/geekOutV2/geekOutV2.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+import completionBackground from "./image/completion.png";
+import runnerUpBackground from "./image/runnerUp.png";
+import { css, Global } from "@emotion/core";
+import styled from "@emotion/styled";
+
+const Container = styled.div`
+  font-family: "Lucida Sans Unicode", "Lucida Grande", sans-serif;
+  width: 297mm;
+  height: 209.9mm;
+  position: relative;
```
Comment (simboonlong) [inline src/templates/geekOutV2/geekOutV2.tsx:15]: 210mm? 🤔

File: src/templates/index.tsx (hunk 1 above)
Comment (yapyuyou) [reply src/templates/index.tsx:13]: yeah the styling and the fields, wanted to extract the 'message' field for future changes

File: src/templates/geekOutV2/geekOutV2.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+      return championBackground;
+    case "runner-up":
+      return runnerUpBackground;
+    case "completion":
+      return completionBackground;
+  }
+};
+
+const processInputDate = (input: string): string => {
+  try {
```
[the remark this answers is not in the document]
Comment (yapyuyou) [reply src/templates/geekOutV2/geekOutV2.tsx]: added geekoutv2.test

File: src/templates/geekOutV2/geekOutV2.tsx (hunk 3 above)
Comment (yapyuyou) [reply src/templates/geekOutV2/geekOutV2.tsx]: ok done

File: src/templates/geekOutV2/geekOutV2.tsx (hunk 4 above)
Comment (yapyuyou) [reply src/templates/geekOutV2/geekOutV2.tsx]: :O not sure why, changed it to 210mm

File: src/templates/index.tsx (hunk 1 above)
Comment (simboonlong) [reply src/templates/index.tsx:13]: now then i noticed about the `message` is part of the background image... disaster

File: src/templates/geekOutV2/types.tsx (hunk 2 above)
Comment (yapyuyou) [reply src/templates/geekOutV2/types.tsx]: i think it's ok here? but trying to fix the circleci now

File: src/templates/geekOut/utils.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+export const getBackgroundImage = (documentType: DocumentType, isV2: boolean): any => {
+  switch (documentType) {
+    case "champion":
+      return isV2 ? championBackgroundV2 : championBackground;
+    case "runner-up":
+      return isV2 ? runnerUpBackgroundV2 : runnerUpBackground;
+    case "completion":
+      return isV2 ? completionBackgroundV2 : completionBackground;
+  }
+};
```
Comment (simboonlong) [inline src/templates/geekOut/utils.ts:18]: ahh now ic so they are actually totally different images (thanks to the message in bg image), ok my bad suggestion on my part on refactoring into utils

**Review by simboonlong — APPROVED** [review]

**Pull request merged**